app/components/quick-filter.js

```diff
@@ -20,38 +20,45 @@ export default Component.extend({
       case 'today':
         newStartDate = moment().toISOString();
         newEndDate = moment().toISOString();
+        this.set('disableClear', false);
         break;
 
       case 'tomorrow':
         newStartDate = moment().add(1, 'day').toISOString();
         newEndDate = newStartDate;
+        this.set('disableClear', false);
         break;
 
       case 'this_week':
         newStartDate = moment().startOf('week').toISOString();
         newEndDate = moment().endOf('week').toISOString();
+        this.set('disableClear', false);
         break;
 
       case 'this_weekend':
         newStartDate = moment().isoWeekday('Friday').toISOString();
         newEndDate = moment().isoWeekday('Sunday').toISOString();
+        this.set('disableClear', false);
         break;
 
       case 'next_week':
         newStartDate = moment().isoWeekday('Monday').add(1, 'week').toISOString();
         newEndDate = moment().isoWeekday('Sunday').add(1, 'week').toISOString();
+        this.set('disableClear', false);
         break;
 
       case 'this_month':
         newStartDate = moment().startOf('month').toISOString();
         newEndDate = moment().endOf('month').toISOString();
+        this.set('disableClear', false);
         break;
 
       default:
     }
 
     this.set('startDate', newStartDate);
     this.set('endDate', newEndDate);
+
   },
 
   init() {
```

Comment:
There are several problems with your solution:

- `init()` is executed only once: when the component loads for the very first time on the page. When the page is loaded for the very first time, these fields will already be null. So this change is irrelevant. 

- If a user clicks on the text field, doesn't type anything. And clicks on the filter/search button. Then an empty `?event_name=` is appended to the URL. It was possible to clear it using the clear filter button. After the changes in this PR, it is not possible. You need to take care of that edge case.

- You need to squash your commits.